libdnf/module/module_db.hpp

```diff
@@ -0,0 +1,112 @@
+/*
+Copyright Contributors to the libdnf project.
+
+This file is part of libdnf: https://github.com/rpm-software-management/libdnf/
+
+Libdnf is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 2.1 of the License, or
+(at your option) any later version.
+
+Libdnf is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#ifndef LIBDNF_MODULE_MODULE_DB_HPP
+#define LIBDNF_MODULE_MODULE_DB_HPP
+
+#include "system/state.hpp"
+
+#include "libdnf/module/module_sack.hpp"
+
+#include <string>
+#include <utility>
+#include <vector>
+
+namespace libdnf::module {
+
+
+class ModuleDB {
+public:
+    ModuleDB(libdnf::system::State & system_state, ModuleSackWeakPtr module_sack)
+        : system_state(system_state),
+          module_sack(module_sack) {}
```

Comment:
I have a proposal to resolve the init issue. What about to replace the arguments by Base argument (weak_ptr)?
Correct me if I am wrong, Module_sack is only used to get Base. And the system state could be received from Base. What do you think?

Reply:
Oh, right. My idea was to access it via `ModuleSack`, but you are right it could be replaced by `Base` weak pointer. I'll make the change.